presto-spi/src/main/java/com/facebook/presto/spi/statistics/TableStatistics.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.spi.statistics;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public final class TableStatistics
+{
+    public static final TableStatistics EMPTY_STATISTICS = TableStatistics.builder().build();
+
+    private final Optional<StatisticsValue<Long>> rowsCount;
```

Comment:
Yes, but with `NaN` you can just write `a + b` and `NaN` will propagate. With optionals you have to write:
```
if (a.isPresent() && b.isPresent()) {
  return Optional.of(a.get() + b.get());
} else {
  return Optional.empty();
}
```
for more complex formulas it is even more cumbersome